File: cpp/Board.cpp
#include "Board.hpp"

Board::Board(uint64_t size)
    : size{size}
{
    if (0 >= size) {
        throw std::runtime_error("Please provide a positive integer board size.");
    }
    this->board = std::vector(this->size, std::vector<bool>(this->size, false));
}

bool Board::isSet(uint64_t row, uint64_t col) const
{
    if (row < 0 || row >= this->size || col < 0 || col >= this->size) {
        throw std::runtime_error(std::format("{} or {} are out of bounds for board of size {}.", row, col, this->size));
    }
    return this->board.at(row).at(col);
}

void Board::set(uint64_t row, uint64_t col)
{
    if (row < 0 || row >= this->size || col < 0 || col >= this->size) {
        throw std::runtime_error(std::format("{} or {} are out of bounds for board of size {}.", row, col, this->size));
    }
    this->board.at(row).at(col) = true;
}

const uint64_t& Board::getSize() const {
    return this->size;
}

File: java/src/main/java/game_of_life/main/Board.java
package game_of_life.main;

import java.text.MessageFormat;

public class Board
{
    public final int size;
    private boolean[][] b;

    public Board(int size) {
        if (!(0 < size)) {
            throw new NegativeArraySizeException("Please provide a positive integer board size.");
        }

        this.size = size;
        this.b = new boolean[size][size];
        for (var row : this.b) {
            for (var i : row) { 
                i = false; 
            }
        }
    }

    public void set(int row, int col) {
        if (row < 0 || row >= this.size || col < 0 || col >= this.size) {
            throw new ArrayIndexOutOfBoundsException(MessageFormat.format("{0} or {1} are out of bounds for board of size {2}.", row, col, this.size));
        }
        this.b[row][col] = true;
    }

    public boolean isSet(int row, int col) {
        if (row < 0 || row >= this.size || col < 0 || col >= this.size) {
            throw new ArrayIndexOutOfBoundsException(MessageFormat.format("{0} or {1} are out of bounds for board of size {2}.", row, col, this.size));
        }
        return this.b[row][col];
    }
}